python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,706 @@
+        self._init_global_eid_offsets()
+        self._init_ntype_maps()
+        self._init_etype_maps()
+        self._init_ntype_features()
+        self._init_etype_features()
+        self._init_etype_files()
+        self._init_ntype_feature_files()
+        self._init_etype_feature_files()
+
+    def __load_json(self, file_name, base_dir):
+        """Helper function to read the metadata schema file of the input
+        graph dataset. ``file_name`` input argument can be either in absolute
+        or relative path formats. ``base_dir`` is used as the root directory
+        for the input graph dataset where all the files (of the dataset) are
+        expected to be located on the disk.
+
+        Argument
+        --------
+        file_name: string
+            Filename of the input metadata schema file. This can be either
+            in absolute or relative path formats. If in relative format,
+            ``base_dir`` is used as the root directory.
+        base_dir: string
+            Base directory, if provided, is used as the root directory for
+            the input graph dataset
+        """
+        self._basedir = None
+        if base_dir is None:
+            self._basedir = os.getcwd()
+        else:
+            self._basedir = base_dir
+
+        self._metadata_path = os.path.join(
+            os.path.join(self._basedir, file_name)
+        )
+        with open(self._metadata_path, "r", encoding="utf-8") as handle:
+            self._data = json.load(handle)
+
+    def __validate_metadata(self):
+        """Helper function to validate the input metadata schema contents.
+        Following set of rules are encoforced to ensure the correctness of
+        the input metadata schema file.
+           * ``graph_name`` should be a valid string,
+           * ``num_nodes_per_type`` should be a valid list of integers
+           * ``num_edges_per_type`` should be a valid list of integers
+           * ``node_type`` is a list whose length should be same as the
+                length of the list ``num_nodes_per_type``
+           * ``edge_type`` is a list whose length should be same as the
+                length of the list ``num_edges_per_type``
+           * No. of node types with node features should be less then
+                total no. of node types
+           * No. of edge types with edge features should be less than
+                total no. of edge types
+           * Node type with node features should be a valid node type,
+                valid node types are in the ``node_type`` list
+           * Edge type with edge features should be a valie edge type,
+                valid edge types are in the ``edge_type`` list
+        """
+        assert isinstance(self._data[STR_GRAPH_NAME], str), "Invalid graph name"
+        assert len(self._data[STR_GRAPH_NAME]) > 0, "Invalid graph name"
+
+        assert (
+            len(self._data[STR_NUM_NODES_PER_TYPE]) > 0
+        ), "Invalid number of nodes"
```

Comment:
do we need the bracket? please follow the convention of other code.

Reply:
it seems your own coding style is not consistent. sometimes you have bracket and sometimes you don't use bracket.